### ISSUE TITLE
Support 'continued train' in C API

### ISSUE THREAD

**Issue by Anstinus** [microsoft/LightGBM#4814]
C API does not have a proper interface to support 'Continued train'. Please consider adding it.

**Comment by shiyu1994** [comment]
@Anstinus Thanks for using LightGBM. I think with C API, we can still use the `.init` file to specify initial scores when continued training. Would you please refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#continued-training-with-input-score to see if that works in your case?

**Comment by Anstinus** [comment]
> @Anstinus Thanks for using LightGBM. I think with C API, we can still use the `.init` file to specify initial scores for continued training. Would you please refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#continued-training-with-input-score to see if that works in your case?

1. Does 'initial score' yield the same result as 'continued train'? 
2. As far as I understood, if I trained 1000 iteration the first time and continued trained for another 1000 iterations, I would get 2000 iterations in the final saved model. So, if I use the 'initial score' way, I suppose I would get two files with 1000 iterations in each and need to manually merge them?

**Comment by shiyu1994** [comment]
@Anstinus Yes. With `initial score` there will be two model files. For inference, we need to predict with the first model first. And use the outputs as the initial scores when predicting with the second model.

I haven't checked why input model on C API is not implemented. I'll come back when I have findings.
